src/components/CreateTeam/CreateTeamForm/index.jsx

```diff
@@ -1,64 +1,118 @@
 import React from 'react';
 import { PropTypes as PT } from 'prop-types';
 import container from './container';
-import { Form, Input } from '../../common';
+
 import { FormContainer } from './styled';
+import { FormStyleContainer } from '../../common_styled/FormStyleContainer';
+import { withFormik } from 'formik';
+
+
+const FormikEnhancer = withFormik({
+  displayName: 'Create Team Form',
+
+  // This updates the form when props change.
+  enableReinitialize: true,
+
+  // validateOn
+  mapPropsToValues: () => ({
+    selectedClient: -1,
+    teamName: '',
+  }),
+
+  // Custom sync validation
+  validate: ({ teamName, selectedClient }) => {
+    let errors = {};
+    if (teamName === '') {
+      errors.teamName = 'Team name cannot be empty.';
+    }
+    if (parseInt(selectedClient) === -1) {
+      errors.selectedClient = 'Please select a client';
+    }
+    return errors;
+  },
+
+  handleSubmit: (payload, bag) => {
+    bag.props.handleFormSubmit(payload);
+    bag.resetForm();
```

Comment:
Might be better to pass the reset form function on through to handleFormSubmit and handle the reset when the creation of the team was successful.

Reply:
I had a feeling this might be mentioned. just means I have to go up three levels, to come back down three levels. Or I might be able to do a componentDidUpdate on the parent container. I'll take a look.